sourmash/tenx.py

```diff
@@ -122,14 +122,16 @@ def shard_bam_file(bam_file_path, chunked_file_line_count):
     startt = time.time()
     file_names = []
 
+    temp_folder = tempfile.mkdtemp()
+
     with read_bam_file(bam_file_path) as bam_file:
         line_count = 0
         file_count = 0
         header = bam_file.header
         for index, alignment in enumerate(bam_file):
             if line_count == 0:
                 file_name = os.path.join(
-                    "/tmp/",
+                    temp_folder,
```

Comment:
I think we should avoid hardcoding this, and in fact it's probably best to have it within the current directory - often `/tmp/` has limited space, compared to cwd. There may be an appropriate "standard" UNIX-y way to allow users to specify this with a TMP variable or some such, but if so I don't know what that is :)

Reply:
The reason why it was not /tmp/ folder and the current directory was exactly that /tmp/ had limited space

Reply:
aaaaaand I think the current code is good. My apologies for the confusion!

Reply:
I suggest using a context manager (starting in line 125), like `with tempfile.TemporaryDirectory() as temp_folder:`, because it will do the cleanup after everything is executed (instead of waiting for the OS to clean it).

More info: https://pymotw.com/3/tempfile/#temporary-directories

Reply:
that's a good point, @luizirber ! thx!

Reply:
good points. added an argument shard_folder to save the sharded bam files to. so, its clear where they get saved. If its in the /tmp/ folder, they are deleted